### PR TITLE
[FIX] res.user: global rule doesn't appear in the list

### PR DESCRIPTION
**Pull request by fmdl**
When you click on the button Rules in res.users form, global rules is not show. (The rules count is not good).

@tde-banana-odoo 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
cc @odony

**Comment by fmdl** [comment]
@dbeguin it is a fix for this work https://github.com/odoo/odoo/pull/25587

File: odoo/addons/base/models/res_users.py [hunk 1]
```diff
@@ -359,10 +359,15 @@ def _compute_tz_offset(self):
 
     @api.depends('groups_id')
     def _compute_accesses_count(self):
+        IrRule = self.env['ir.rule']
```
Comment (dbeguin) [inline odoo/addons/base/models/res_users.py]: if used only once, no need to make a variable for this ;) 
while we are at it, we could externalize this in a method _getRules or something, as it's used twice in same file.

Comment (fmdl) [reply odoo/addons/base/models/res_users.py]: @dbeguin done

**Comment by odony** [comment]
I'm really not too sure about this. The group rules are part of our role-based access control configuration, but global rules are not role-based and are more like hard-coded behaviors. They're generally only used for multi-company isolation, and they apply to all users indifferently.
I find misleading to show them in a role-based context. There's also little chance that they will need to be modified for a given user, or that reviewing them via the Rules button will allow an admin to spot a mistake or security risk.

PS: there is a similar weirdness with ACL lines without a group, yet those could be considered part of the ACL config and could indicate security problems...

**Comment by fmdl** [comment]
@odony I understand your point of view, this is not the best.

On ACL a warning is added (but on ir.rules).
image
Global = all groups, that why I think it better to show them. 

With https://github.com/odoo/odoo/pull/29122, my idea is to make a better link between user and ACLs/rules.
But today groups are between user and ACLs/rules, and it is hard to check.
- If you go on a group to modify ACLs/rules, it is not easy because you need to go on each groups of the user.
- If you go on ACLs/rules list, you need to remember each groups of the users

Maybe we need to work on a better "security center" (view, java view, wizard and/or pdf report) with two main feature :
- check what user have access : list of model with percentage of access (10 record can read of 100 total records : 10%). This can help to make an audit. (exemple : who have access to invoice ?)
- show ACLs/rules for one user and one model (to solve a user call : "I can't access to blabla"), to check quickly witch ACLs or rules are concerned (and not navigate between groups/ACLs/rules)

Note : some groups give extra right (and some time is not justify, exemple Warehouse Manager give to access to read Invoice)), and it is hard to check if everything is ok. 
It could be nice to create `check rules` (like a test security), exemple:
-`User A` `can not` `read` `invoice A`. 
-`User A` or `User B` `can not` `create` `Invoice`.
With this check rule, if you add a new module, change groups of user or make a migration, the check rules is tested and raise (or send email) if need. I know that it is a fin tunning, but for a big company it is an insurance that the ACLs and rules are well configured.

ACLs and rules is a big brain teaser (it is the only on Odoo :-) ), it is hard to check if user have access and it is hard to ensure that a user have never access to some document (accounting for exemple).

Tell me your point of view.

**Comment by fmdl** [comment]
@odony I understand your point of view, but somes rules is not only multi-company (like private event).
This idea is : an user call because he don't see some invoice, if it because a global rules, you don't understand. (go to user, show rules, show only account.invoice model).

Global = all groups, that why I think it better to show them. 

On ACL a warning is added (but on ir.rules).
image

With https://github.com/odoo/odoo/pull/29122, my idea is to make a better link between user and ACLs/rules.
But today groups are between user and ACLs/rules, and it is hard to check.
- If you go on a group to modify ACLs/rules, it is not easy because you need to go on each groups of the user.
- If you go on ACLs/rules list, you need to remember each groups of the users

Maybe we need to work on a better "security center" (view, java view, wizard and/or pdf report) with two main feature :
- check what user have access : list of model with percentage of access (10 record can read of 100 total records : 10%). This can help to make an audit. (exemple : who have access to invoice ?)
- show ACLs/rules for one user and one model (to solve a user call : "I can't access to blabla"), to check quickly witch ACLs or rules are concerned (and not navigate between groups/ACLs/rules)

Note : some groups give extra right (and some time is not justify, exemple Warehouse Manager give to access to read Invoice)), and it is hard to check if everything is ok. 
It could be nice to create `check rules` (like a test security), exemple:
-`User A` `can not` `read` `invoice A`. 
-`User A` or `User B` `can not` `create` `Invoice`.
With this check rule, if you add a new module, change groups of user or make a migration, the check rules is tested and raise (or send email) if need. I know that it is a fin tunning, but for a big company it is an insurance that the ACLs and rules are well configured.

ACLs and rules is a big brain teaser (it is the only on Odoo :-) ), it is hard to check if user have access and it is hard to ensure that a user have never access to some document (accounting for exemple).

Tell me your point of view.

**Comment by fmdl** [comment]
ping @odony

**Comment by odony** [comment]
As I said, not a fan. See with @dbeguin and @tde-banana-odoo if they consider it useful from a functional point of view ;)

IMO with #25587 and this PR we've been bloating the UI and code with redundant stuff.

If the admin knows the ACL system very well, and the models and their relationships, then the buttons are just redundant info from the regular ACL menus. And if the admin doesn't, same result as with the basic menus: they don't understand anything.

I'm also concerned about the extra queries that are now added every time we open a user form. Those counts (_X Access Rights_, _Y Record Rules_) don't seem to provide any useful info, do they?

**Pull request closed without merging**